src/transaction/boot_sr.c

```diff
@@ -2248,7 +2248,12 @@ boot_restart_server (THREAD_ENTRY * thread_p, bool print_restart, const char *db
       goto error;
     }
 
-  init_server_type (db_name);
+  error_code = init_server_type (db_name);
+  if (error_code != NO_ERROR)
+    {
+      // error already set
```

Comment:
When error is already set, use `ASSERT_ERROR ()`
```suggestion
      ASSERT_ERROR ();
```

Reply:
Done